nixos/modules/services/misc/memos.nix

```diff
@@ -0,0 +1,197 @@
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}:
+let
+  cfg = config.services.memos;
+  envFileFormat = pkgs.formats.keyValue { };
+in
+{
+  options.services.memos = {
+    enable = lib.mkEnableOption "Memos note-taking";
+    package = lib.mkPackageOption pkgs "Memos" {
+      default = "memos";
+    };
+
+    openFirewall = lib.mkEnableOption "opening the ports in the firewall";
+
+    user = lib.mkOption {
+      type = lib.types.str;
+      description = ''
+        The user to run Memos as.
+
+        ::: {.note}
+        If changing the default value, **You** are responsible of creating the corresponding user with {option}`users.users`.
+        :::
+      '';
+      default = "memos";
+    };
+
+    group = lib.mkOption {
+      type = lib.types.str;
+      description = ''
+        The group to run Memos as.
+
+        ::: {.note}
+        If changing the default value, **You** are responsible of creating the corresponding group with {option}`users.groups`.
+        :::
+      '';
+      default = "memos";
+    };
+
+    dataDir = lib.mkOption {
+      default = "/var/lib/memos/";
+      type = lib.types.path;
+      description = ''
+        Specifies the directory where Memos will store its data.
+
+        ::: {.note}
+        It will be automatically created with the permissions of {option}`services.memos.user` and {option}`services.memos.group`.
+        :::
+      '';
+    };
+
+    settings = lib.mkOption {
```

Comment:
Great to see you found a way to implement RFC42-style freeform structured settings! It's a shame they're env vars based on CLI options, but at least it works.